File: source-connector/src/main/java/sourceconnector/repository/file/decompressor/GzipDecompressor.java
package sourceconnector.repository.file.decompressor;

import offsetmanager.domain.file.FileKey;

import java.io.IOException;
import java.io.InputStream;
import java.util.zip.GZIPInputStream;

public class GzipDecompressor implements Decompressor {
  @Override
  public boolean supports(FileKey fileKey) {
    return fileKey.get().endsWith(".gz");
  }

  @Override
  public InputStream decompress(InputStream in) throws IOException {
    return new GZIPInputStream(in);
  }
}

File: source-connector/src/main/kotlin/sourceconnector/repository/file/decompressor/GzipDecompressor.kt
package sourceconnector.repository.file.decompressor

import offsetmanager.domain.file.FileKey
import java.io.IOException
import java.io.InputStream
import java.util.zip.GZIPInputStream

class GzipDecompressor : Decompressor {
  override fun supports(fileKey: FileKey): Boolean {
    return fileKey.get().endsWith(".gz")
  }

  @Throws(IOException::class)
  override fun decompress(`in`: InputStream): InputStream {
    return GZIPInputStream(`in`)
  }
}
